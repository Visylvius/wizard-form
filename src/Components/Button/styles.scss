@import '../../variables';

.Button-component {
  background: $button-background;
  border: 1px solid $button-background;
  border-radius: 3px;
  color: $button-color;
  cursor: pointer;
  height: 35px;
  outline: none;
  width: 199px;

  &:disabled {
    background: $button-background;
    cursor: not-allowed;
  }

  &:hover {
    background: $nps-background;
  }
}

.Button-container {
  display: flex;
  justify-content: center;
  width: 100%;
}
