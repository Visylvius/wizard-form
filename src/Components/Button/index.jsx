import React from 'react';
import classNames from 'classnames';

import './styles.scss';

const Button = ({ buttonClasses, children, shouldCenter, ...rest }) => {
  const buttonClassNames = classNames('Button-component', buttonClasses);

  return (
    <>
      {shouldCenter ? (
        <div className='Button-container'>
          <button className={buttonClassNames} {...rest}>
            {children}
          </button>
        </div>
      ) : (
        <button className={buttonClassNames} {...rest}>
          {children}
        </button>
      )}
    </>
  );
};

export default Button;
