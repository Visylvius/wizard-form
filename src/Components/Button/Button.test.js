import React from 'react';
import renderer from 'react-test-renderer';

import Button from './index';

describe('render', () => {
  test('should render properly', () => {
    const tree = renderer.create(<Button>Hello</Button>);

    expect(tree).toMatchSnapshot();
  });
});
