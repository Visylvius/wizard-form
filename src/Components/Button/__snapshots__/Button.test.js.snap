// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render properly 1`] = `
<button
  className="Button-component"
>
  Hello
</button>
`;
